extra/changelog-generator/changelog-generator

```diff
@@ -148,14 +148,20 @@ for repo in repos:
                 title = line
                 title_fetched = True
 
-            match = re.match("^ *Changelog: *(.*)", line, re.IGNORECASE)
-            if match:
+            match_changelog = re.match("^ *Changelog: *(.*)", line, re.IGNORECASE)
+            if match_changelog:
                 if log_entry:
                     add_entry(sha, log_entry)
                     log_entry = ""
                 log_entry_local = False
 
-                if re.match("^(Title|Commit|None)[ .]*$", match.group(1), re.IGNORECASE):
+                # The logic of this regex is:
+                # - Shall start with Title|Commit|All|None
+                # - From here, if [.:]+ then capture the rest in a new group
+                # - Else just expect end line
+                match_keyword = re.match("^(Title|Commit|All|None) *([.:]+)? *((?(2).*$|$))", match_changelog.group(1), re.IGNORECASE)
```

Comment:
I actually have not seen the `(?(` notation before, I had to look it up.

Question is, is this an easier way to express the same thing?

`^(Title|Commit|All|None) *(?:[.:]+ *(.*)$|$))`

If it doesn't work, I'm fine with the original expression.

Reply:
Your regex also works (all tests pass, which is all the evidence I need). Thank you!